xcube_cmems/cmems.py

```diff
@@ -18,28 +18,43 @@
 # LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
 # OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
 # SOFTWARE.
+import aiohttp
 import asyncio
 from functools import cache
+from io import BytesIO
 from typing import List, Dict, Any, Optional
+import lxml.etree as etree
 import os
 import logging
+import time
 
 import nest_asyncio
-from urllib.parse import urlsplit
 from pydap.cas.get_cookies import setup_session
-from owslib.fes import SortBy
-from owslib.fes import SortProperty
-from owslib.csw import CatalogueServiceWeb
 
 from .constants import CAS_URL
 from .constants import ODAP_SERVER
 from .constants import DATABASE
 from .constants import CSW_URL
-from .constants import TOTAL_RECORDS_CSW
-from .constants import STEP_SIZE
 
 _LOG = logging.getLogger('xcube')
 
+CSW_NAMESPACES = {
+    'dc': 'http://purl.org/dc/elements/1.1/',
+    'csw': 'http://www.opengis.net/cat/csw/2.0.2'
+}
+
+_RECORDS_PER_REQUEST = 25
+_GET_RECORDS_REQUEST = {
+    'service': 'CSW',
+    'request': 'GetRecords',
+    'version': '2.0.2',
+    'resultType': 'results',
+    'ElementSetName': 'full',
+    'maxRecords': _RECORDS_PER_REQUEST
+}
+# Ideally, we would read this from cmems
+_TOTAL_NUM_OF_RECORDS = 300
```

Comment:
I still don't understand why this constant is needed.
I consider this implementation to be wrong.